string_manipulation.rb

```diff
@@ -243,6 +256,9 @@ def palindrome_check(my_phrase)
   return true
 end
 
+# Time complexity is O(n), the loop will run n times depending on the size of my_phrase
+# Space complexity is O(1), no new space in memory is being allocated.
+
 # A method that updates the string by replacing consecutive repeating characters
 # with a number representing the frequency. The replacement is done only if the
 # string length will get reduced by the process.
```

Comment:
Two things:
1. Just like all the other methods in this homework assignment, this method should also be updating the input string in place. So, don't use new_string.
2. Once a non repeating character is found, you want to add the count right after the first occurrence of the character if the count is greater than 2, otherwise, you want to copy those set of characters (one or two) as is.

Hint: In my approach, I found it useful to keep track of where I'm counting and where I'm copying using two different index variables.